### ISSUE TITLE
Enhancement: Add plugin to support image generation using Flux AI API

### ISSUE THREAD

**Issue by kneelesh48** [danny-avila/LibreChat#3671]
### What features would you like to see added?

I'd like to request a plugin that supports image generation using Flux Pro API on replicate and fal.ai

### More details

The below links can be used to send API requests

https://fal.ai/models/fal-ai/flux-pro
https://replicate.com/black-forest-labs/flux-pro

### Which components are impacted by your request?

Plugins

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kociol21** [comment]
Great idea, but plugins are kinda a dead end road, because it works only with OpenAI models. It would be better if there was an option to configure our Flux API (e.g. FalAI) and it would work with any model, for example after command /img or /flux or something like that. Though I think some similar system is being worked on.

**Comment by danny-avila** [comment]
> Great idea, but plugins are kinda a dead end road, because it works only with OpenAI models.

You're right, they will be deprecated for a Tool system that is more modular, and open to every endpoint, see the discussion on this: https://github.com/danny-avila/LibreChat/issues/3607

> It would be better if there was an option to configure our Flux API (e.g. FalAI) and it would work with any model, for example after command /img or /flux or something like that. Though I think some similar system is being worked on.

Also great idea, and it's somewhat related to the open issue I linked. This is very closely aligned to something else I have cooking :)

---

Moving to discussion because something like this is planned with a more general approach, to allow any image generation api to work,
